nixos/modules/services/networking/pounce.nix

```
@@ -0,0 +1,224 @@
+      };
+
+      sslKey = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        example = "/etc/letsencrypt/cert/private.key";
+        description = ''
+          Path to server TLS certificate key.
+        '';
+      };
+
+      sslCert = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        example = "/etc/letsencrypt/cert/cert.crt";
+        description = ''
+           Path to server TLS certificate.
+        '';
+      };
+
+      certFP = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        example = "/etc/pounce/libera-user.pem";
+        description = ''
+           Path to certFP certificate for login to IRC server
+        '';
+      };
+
+      # If you want to use password, you need to generate it with pounce -x
+      hashedPass = mkOption {
+        type = types.str;
+        default = "";
+        description = ''
+        '';
+      };
+
+      ircHost = mkOption {
+        type = types.str;
+        example = "irc.libera.chat";
+        default = "";
+        description = ''
+          IRC channel
+        '';
+      };
+
+      ircHostPort = mkOption {
+        type = types.int;
+        default = 6697;
+        description = ''
+        '';
+      };
+
+      joinChannels = mkOption {
+        type = types.listOf types.str;
+        default = [];
+        description = ''
+          List of channel to join at startup
+        '';
+      };
+
+      ircUser = mkOption {
+        type = types.str;
+        default = "";
+        description = ''
+          IRC user
+        '';
+      };
+
+      ircNick = mkOption {
+        type = types.str;
+        default = "";
+        description = ''
+          IRC nickname
+        '';
+      };
+
+      ircPass = mkOption {
+        type = types.str;
+        default = "";
+        description = ''
+          IRC password
+        '';
+      };
+
+      extraConfig = mkOption {
+        type = types.lines;
+        default = "";
+        description = "Extra configuration";
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    systemd.services.pounce = {
+      description = "Pounce";
+      after = [ "network-online.target" ];
+      wantedBy  = [ "multi-user.target" ];
+      serviceConfig = {
+        User = cfg.user;
+        Group = cfg.group;
+        ExecStart = "${pkgs.pounce}/bin/pounce ${configFile}";
+        Restart = "always";
+        RuntimeDirectory = "pounce";
+        RuntimeDirectoryMode = "0700";
+      };
+    };
+
+    users.users = optionalAttrs (cfg.user == defaultUser) {
+      ${defaultUser} =
+        { description = "Pounce server daemon owner";
+          group = cfg.group;
+          home = cfg.dataDir;
+          isNormalUser = true;
+          createHome = true;
+        };
+      };
+
+    users.groups = optionalAttrs (cfg.user == defaultUser) {
+      ${defaultUser} =
+       {
+         members = [ defaultUser ];
+       };
+    };
+  };
```

```suggestion
    users.groups.${defaultUser} = mkIf (cfg.user == defaultUser) {};
```
setting `group` on a user automatically makes that user a member of the given group